src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -1519,14 +1498,55 @@ void AlgorithmPaintToolBox::onViewClosed()
     }
 }
 
+void AlgorithmPaintToolBox::clearMask()
+{
+    if ( m_maskData && m_itkMask )
+    {
+        m_itkMask->FillBuffer( MaskPixelValues::Unset );
+        m_itkMask->Modified();
+        m_itkMask->GetPixelContainer()->Modified();
+        m_itkMask->SetPipelineMTime(m_itkMask->GetMTime());
+
+        m_maskAnnotationData->invokeModified();
+
+        if(maskHasBeenSaved)
+        {
+            m_imageData->removeAttachedData(m_maskAnnotationData);
+            maskHasBeenSaved = false;
+            setData(currentView->layerData(0));
+        }
+        m_applyButton->setDisabled(true);
+    }
+    m_imageData = NULL;
```

Comment:
I just added this line to `clearMask()`. The function has been moved to be with its related functions.